### PR TITLE
feat(backend): Update config API

### PR DESCRIPTION
**Pull request by Vaish24navi**
## Description
- Built the API for updating configs.
- Added the essential tests for the APIs - covering both unit and integration.

## Related Issue

- [x] Link to the related issue (if applicable) #17 

## Type of Change

- [ ] Bugfix
- [X] New Feature
- [ ] Documentation

## Checklist

- [X] Code follows the project style guidelines
- [X] Tests have been added/updated
- [X] All tests pass

## Release Notes

- **New Features**
  - Introduced an endpoint for updating configurations via HTTP PUT requests.
  - Added a new class for representing configuration updates, enhancing model definitions.
  - Implemented validation for configuration updates to ensure data integrity.

- **Bug Fixes**
  - Improved error handling for configuration updates, ensuring appropriate HTTP responses.

- **Tests**
  - Added new integration and unit tests to validate the configuration update functionality, including various scenarios and edge cases.
  - Expanded test cases in the payloads to cover different update scenarios.

### CODE REVIEW

File: backend/app/api/v1/config.py [hunk 1]
```diff
@@ -136,3 +136,34 @@ def list_configs(config_definition_key: str, page: int = 1, limit: int = 10):
...
+            config_definition_key,
+            config_key,
+            config.data,
+        )
+    except ValueError as e:
+        logger.warning(f"Configuration not found: {e}")
+        raise HTTPException(status_code=404, detail=str(e))
+    except Exception as e:
+        logger.exception(f"Error updating configuration: {e}")
+        raise HTTPException(status_code=500, detail=str(e))
```
[the remark this answers is not in the document]
Comment (niyasrad) [reply backend/app/api/v1/config.py:167]: I respectfully disagree with this, since we will be making a refactor wave once the set of APIs are validated and tested.

File: backend/app/utils/configs/utils.py [hunk 2]
```diff
@@ -121,3 +123,29 @@ def l_config(config_definition_key: str, page: int = 1, page_size: int = 10):
...
+
+    validate_config_update(config_definition_key, config_key, data)
+
+    update_query, update_params = u_config_query(
+        config_definition_key, config_key, data
+    )
+    rows_affected = data_store.execute_query(update_query, update_params)
+
+    if rows_affected == 0:
+        raise ValueError(f"Configuration with key '{config_key}' not found.")
```
[the remark this answers is not in the document]
Comment (niyasrad) [reply backend/app/utils/configs/utils.py:149]: We will take this up!

File: backend/tests/unit_tests/config/payloads/payloads.json [hunk 3]
```diff
@@ -116,7 +116,42 @@
...
+            "rows_affected": 0
+        },
+        "test_update_o_key": {
+            "config_definition_key": "sample_config_definition",
+            "config_key": "",
+            "data": {"key1": "value1", "key2": "value2"},
+            "schema": {"type": "object", "properties": {"key1": {"type": "string"}, "key2": {"type": "string"}}},
+            "rows_affected": 0
+        }
+    },
```
[the remark this answers is not in the document]
Comment (niyasrad) [reply backend/tests/unit_tests/config/payloads/payloads.json:154]: We will also take this up!

File: backend/app/api/v1/config.py [hunk 4]
```diff
@@ -136,3 +136,34 @@ def list_configs(config_definition_key: str, page: int = 1, limit: int = 10):
         "message": "Configurations listed successfully.",
         "data": configs,
     }
+
+
```
Comment (niyasrad) [inline backend/app/api/v1/config.py:140]: Let's place the API in it's appropriate place, in utils, validations, api!

File: backend/tests/unit_tests/config/payloads/payloads.json [hunk 5]
```diff
@@ -170,47 +237,46 @@
...
+        "page": 1,
+        "page_size": -1
+      }
+    },
+    "test_list_n_plimit": {
+      "config_definition_key": "sample_config_definition",
+      "data": {
+        "page": 1,
+        "page_size": 101
       }
```
[the remark this answers is not in the document]
Comment (niyasrad) [reply backend/tests/unit_tests/config/payloads/payloads.json:279]: I guess this could be done by using some new assertions like

```
if expect_error:
    with pytest.raises(ValueError) as exc_info:
        u_config_definition(config_key, updated_indexes_list)
    assert str(exc_info.value) == "Invalid request: page_size must be less than or equal to 100"
    mock_execute_query.assert_not_called()
    return
```

Can you make a new issue for this? This can be called `tests(general): adding ValueError message assertions`?

**Comment by niyasrad** [comment]
Wonderful work @Vaish24navi 🥂
